File: Workshop 3/src/BlackJack/view/SimpleView.java
/*
 * SimpleView.java
 * Date: 26 Oktober, 2014
 * Author: Tobias Ohlsson (edited by Robert Willem Hallink)
 */
package BlackJack.view;

/*
 * Mostly left untouched.
 * Added three new display messages and an extra line print.
 * Not a change however, but GetInput is not used anymore.
 * Is now extended by AView which implements IView.
 */
public class SimpleView extends AView  {
    
    //Two new display methods, for better user feedback.
    @Override   public void DisplayStartup()        { System.out.print("Type in a letter and press enter to start: "); }
    @Override   public void DisplayEnterInput()     { System.out.print("Enter input: "); }
    @Override   public void DisplayInvalidInput()   { System.out.println("Unknown input, try again please."); }
    @Override   public void DisplayLastInput()      { System.out.print("Last input: "); }
    
    @Override
    public void DisplayWelcomeMessage() {
        
        for (int i=0; i<50; i++)
            System.out.print("\n");
        
        System.out.println("Hello Black Jack World");
        System.out.println("Type 'p' to Play, 'h' to Hit, 's' to Stand or 'q' to Quit\n");
        
    }
    
    @Override
    public void DisplayCard(BlackJack.model.Card a_card) {
        System.out.println("" + a_card.GetValue() + " of " + a_card.GetColor());
    }
    
    @Override
    public void DisplayPlayerHand(Iterable<BlackJack.model.Card> a_hand, int a_score) {
        DisplayHand("Player", a_hand, a_score);
    }
    
    @Override
    public void DisplayDealerHand(Iterable<BlackJack.model.Card> a_hand, int a_score) {
        DisplayHand("Dealer", a_hand, a_score);
    }
    
    private void DisplayHand(String a_name, Iterable<BlackJack.model.Card> a_hand, int a_score) {
        
        System.out.println(a_name + " Has: ");
        for (BlackJack.model.Card c : a_hand)
            DisplayCard(c);
        System.out.println("Score: " + a_score);
        System.out.println("");
        
    }
    
    @Override
    public void DisplayGameOver(boolean a_dealerIsWinner) {
        
        System.out.println("GameOver: ");
        if (a_dealerIsWinner)
            System.out.println("Dealer Won!");
        else System.out.println("You Won!");
        System.out.println();       //Added, for a bit more display value
        
    }
    
}

File: Workshop 3/Files/Peer Review/Materials - Eddy Proca/view/SimpleView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BlackJack.view
{
    class SimpleView : IView
    {
        private static string englishPlayAction = "p";
        private static string englishHitAction = "h";
        private static string englishStandAction = "s";
        private static string englishQuitAction = "q";
        public void DisplayHeadline()
        {
            Console.Clear();
            Console.WriteLine("Black Jack");
            Console.WriteLine();
        }
        public void DisplayWelcomeMessage()
        {
            DisplayHeadline();
            Console.WriteLine("Type '{0}' to play, '{1}' to Hit, '{2}' to Stand or '{3}' to Quit\n", englishPlayAction.ToString(), englishHitAction, englishStandAction, englishQuitAction);
        }
        public controller.PlayGame.gameAction GetInput()
        {
            var input = System.Console.ReadLine();
            if (input == englishPlayAction)
            {
                return controller.PlayGame.gameAction.play;
            }
            else if (input == englishHitAction)
            {
                return controller.PlayGame.gameAction.hit;
            }
            else if (input == englishStandAction)
            {
                return controller.PlayGame.gameAction.stand;
            }
            else if (input == englishQuitAction)
            {
                return controller.PlayGame.gameAction.quit;
            }
            else
            {
                return controller.PlayGame.gameAction.invalid;
            }
        }
        public void DisplayCard(model.Card a_card)
        {
            Console.WriteLine("{0} of {1}", a_card.GetValue(), a_card.GetColor());
        }
        public void DisplayPlayerHand(IEnumerable<model.Card> a_hand, int a_score)
        {
            DisplayHand("Player", a_hand, a_score);
        }
        public void DisplayDealerHand(IEnumerable<model.Card> a_hand, int a_score)
        {
            DisplayHand("Dealer", a_hand, a_score);
        }
        private void DisplayHand(String a_name, IEnumerable<model.Card> a_hand, int a_score)
        {
            System.Console.WriteLine("{0} Has: ", a_name);
            foreach (model.Card c in a_hand)
            {
                DisplayCard(c);
            }
            System.Console.WriteLine("Score: {0}", a_score);
            System.Console.WriteLine("");
        }
        public void DisplayGameOver(bool a_dealerIsWinner)
        {
            System.Console.Write("GameOver: ");
            if (a_dealerIsWinner)
            {
                System.Console.WriteLine("Dealer Won!");
            }
            else
            {
                System.Console.WriteLine("You Won!");
            }
        }
    }
}
